llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1764,6 +1764,10 @@ static VPRecipeBase *createEVLRecipe(VPValue *HeaderMask,
   using namespace llvm::VPlanPatternMatch;
   auto GetNewMask = [&](VPValue *OrigMask) -> VPValue * {
     assert(OrigMask && "Unmasked recipe when folding tail");
+    using namespace llvm::VPlanPatternMatch;
+    VPValue *Mask;
+    if (match(OrigMask, m_LogicalAnd(m_Specific(HeaderMask), m_VPValue(Mask))))
```

Comment:
might be good to add a quick comment with an explantation., e.g. something like
```suggestion
    // HeaderMask will be handled using EVL.
    if (match(OrigMask, m_LogicalAnd(m_Specific(HeaderMask), m_VPValue(Mask))))
```